frontend/src/services/index.js

```diff
@@ -1,15 +1,29 @@
 import axios from "axios";
+import jwtDecode from "jwt-decode";
+import dayJS from "dayjs";
+
 
 const API = axios.create({
 	baseURL: process.env.BASEURL,
 });
 
 if (localStorage.getItem("user")) {
 	const { accessToken } = JSON.parse(localStorage.getItem("user"));
+	const user = jwtDecode(accessToken);
+	const userID = jwtDecode(accessToken).UserID;
+	const isExpired = dayJS.unix(user.exp).diff(dayJS()) < 1;
+
 	API.interceptors.request.use(
 		async (config) => {
 			config.headers.Authorization = `Bearer ${accessToken}`;
-			return config;
+			if(!isExpired) {
+				return config;
+			} else  {
+				const response = await axios.post(`${process.env.BASEURL}/user/refreshtoken/${userID}`)
+				localStorage.setItem("newToken", JSON.stringify(response.data.accessToken))
+				config.headers.Authorization = `Bearer ${response.data.accessToken}`;
+				return config;
+			}
```

Comment:
@Zorin95670 made a comment that you should put that in a separate function (in another file). This PR has been recreated so I guess that's why it does not appear?

Reply:
Discussed with @BrahimS : he think that it should not be put in a separated function. We should not loose too much time on this so I'll validate the PR.